File: Data Structures & Algorithms/binary-search/submission-1.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        
        int right = nums.size()-1, left = 0;

        if(nums[right]==target){
            return right;
        }

        if(nums[left] ==target){
            return left;
        }

        while(left<=right){

            int mid = (right+left)/2;
            
            if (target > nums[mid]){
                left = mid+1;
            }else if (target<nums[mid]){
                right = mid-1;
            }else{
                return mid;
            }
        }
        return -1;



        
    }
};

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:

        if root is None:
            return []

        queue, result = [root], []
        level_size = 1

        while queue:
            cur_node = queue.pop(0)

            if result == []:
                result.append([cur_node.val])
            else:
                if len(result[-1]) == level_size:
                    level_size = len(queue) + 1
                    result.append([cur_node.val])
                else:
                    last_level = result.pop()
                    last_level.append(cur_node.val)
                    result.append(last_level)

            if cur_node.left:
                queue.append(cur_node.left)
            if cur_node.right:
                queue.append(cur_node.right)

        return result